src/v/raft/consensus.cc

```diff
@@ -720,19 +721,39 @@ ss::future<> consensus::start() {
                   _term = lstats.dirty_offset_term;
                   _voted_for = {};
               }
-              vlog(
-                _ctxlog.info,
-                "Recovered, log offsets: {}, term:{}",
-                lstats,
-                _term);
+              auto f = ss::now();
+              /**
+               * Snapshot and log was delted, try to recover by creating
+               * snapshot with initial data.
+               *
+               * MOST IMPORTANT:
+               * we can only do this when log is empty and there
+               * is no snapshot for given offset.
+               */
+              if (
+                _log.segment_count() == 0
+                && lstats.start_offset > lstats.dirty_offset
+                && _last_snapshot_index != lstats.dirty_offset) {
```

Comment:
we should add a comment on the snapshot directory bing in the same directory as the partition